### PR TITLE
Extract interfaces from Rule

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

**Comment by vjik** [comment]
Maybe name interfaces without `Validator`?
`ParametrizedRuleInterface`, `TranslatableRuleInterface` and etc.

File: src/FormatableRuleInterface.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator;
+
+/**
+ * Rule represents a single value validation rule.
```
Comment (samdark) [inline src/FormatableRuleInterface.php:8]:
```suggestion
 * Formattable rule constructs error messages from a pattern and parameters. The interface allows to set formatter to use.
```

File: src/ParametrizedRuleInterface.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator;
+
+/**
+ * Rule represents a single value validation rule.
+ */
+interface ParametrizedRuleInterface extends RuleInterface
```
Comment (samdark) [inline src/ParametrizedRuleInterface.php:10]:
```suggestion
/**
 * The interface should be implemented for a rule that is meant to be converted to array of name => parameters.
 * Such array is usually passed to the client and then used for client-side validation.
 */
interface ParametrizedRuleInterface extends RuleInterface
```

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**